src/test/java/com/puppycrawl/tools/checkstyle/internal/XpathRegressionTest.java

```diff
@@ -193,7 +193,7 @@ public void validateIncompatibleJavadocCheckNames() throws IOException {
         final Set<Class<?>> abstractJavadocCheckNames = CheckUtil.getCheckstyleChecks()
                 .stream()
                 .filter(AbstractJavadocCheck.class::isAssignableFrom)
-                .collect(Collectors.toSet());
+                .collect(Collectors.toCollection(HashSet::new));
         // add the extra checks
         abstractJavadocCheckNames.addAll(REGEXP_JAVADOC_CHECKS);
```

Comment:
we can make abstractJavadocCheckNames immutable and add it to mutable collection that is initialized by REGEXP_JAVADOC_CHECKS

Reply:
The implementation of `REGEXP_JAVADOC_CHECKS` uses `Set#of`. This is the Javadoc of `Set#of`:

>  Returns an unmodifiable set containing four elements. See Unmodifiable Sets for details.
....
Throws:
IllegalArgumentException – if there are any duplicate elements
NullPointerException – if an element is null
Since:
9 static <E> Set<E> of(E e1, E e2, E e3, E e4) {
        return new ImmutableCollections.SetN<>(e1, e2, e3, e4);
    }

So we cannot add the `abstractJavadocCheckNames` to that collection. Should we update the implementation of `REGEXP_JAVADOC_CHECKS` or should we combine the two sets in a way?

Reply:
ok, one unmodifiable is enough, so ok to keep `toCollection(HashSet::new)`